bindings/py/tests/regions/network_test.py

```diff
@@ -258,10 +258,11 @@ def testGetInputArray(self):
     r_from = network.addRegion("from", "py.LinkRegion", "")
     r_to = network.addRegion("to", "py.LinkRegion", "")
     network.link("from", "to", "", "", "UInt32", "UInt32")
+    network.link("INPUT", "from", "", "{dim: [5]}", "UInt32_source", "UInt32")
```

Comment:
I like this, it's quite understandable :+1:

Reply:
Thank you.  I struggled a lot with it.
The third argument is defined in the API as "Link type".  I thought about making this a new link type rather than using "INPUT" to identify the link as one of those direct input things.  But decided against it.

Under the covers there is a new region InputRegion which is identified with the name INPUT. It has no Spec. When "INPUT" is given in a link, an instance of this InputRegion is created and for each unique 'source_name' mentioned an output is created for it.  All part of the Link declaration processing. So the setInputData( ) function actually sets an output buffer that is connected to the target input buffer with the link.  This avoided a lot of complications trying to directly set the input buffer.

Reply:
So this "INPUT" needs to be set only for the 1st region in the chain (typically some Encoder region, FileIORegion, or now DatabaseRegion)? And then it'll help to describe dims for this (actually) 1st region. Other regions derive their dimensions based on their linked I/O? 
Or the "INPUT" needs to be set for each region in the network? (like specifying region's dimensions?)

Reply:
> So this "INPUT" needs to be set only for the 1st region in the chain

No, the "INPUT" link only needs to be set if you want your application to provide data for an input.  If you are using an Encoder, FileInputRegion, or the new DataBaseRegion to provide the data, the data is flowing from their output buffer.  We have always had multiple ways to set the dimensions on outputs.  See "Dimensions and determining buffer size" section of docs/NetworkAPI_Creating_Regions.md

But the INPUT feature needed a way to set the dimensions on the input of the target region rather than on its output.  So I provided a way to configure that with the link parameters.